internal/service/autoscaling/filters.go

```diff
@@ -0,0 +1,55 @@
+package autoscaling
+
+import (
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/autoscaling"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func buildFiltersDataSource(set *schema.Set) []*autoscaling.Filter {
```

Comment:
🎨  unless the functions are being reused, we prefer to keep them as close to the relevant files as possible. these can be moved directly into the data_source file